src/player.cpp

```diff
@@ -14149,12 +14142,28 @@ int player::get_stamina_max() const
 
 void player::burn_move_stamina( int moves )
 {
+    int overburden_percentage = 0;
+    int current_weight = weight_carried();
+    int max_weight = weight_capacity();
+    if (current_weight > max_weight) {
+        overburden_percentage = (current_weight - max_weight) * 100 / max_weight;
+        // Chance to suffer pain if stamina runs out or has trait BADBACK
+        // Starts at 1 in 25, goes down by 5 for every 50% more carried
+        if ((has_trait("BADBACK") || stamina <= 0) && one_in(35 - 5 * current_weight / (max_weight / 2))) {
```

Comment:
This one makes it only trigger when player has BADBACK.
This is because stamina is regenerated before this function is called and then decreased at the end of it, so stamina is usually above 0 here.